packages/bruno-electron/src/ipc/collection.js

```diff
@@ -171,9 +172,13 @@ const registerRendererEventHandlers = (mainWindow, watcher, lastOpenedCollection
         await createDirectory(envDirPath);
       }
 
-      const envFilePath = path.join(envDirPath, `${environment.name}.bru`);
+      let envFilePath = path.join(envDirPath, `${sanitizeFilename(environment.name)}.bru`);
       if (!fs.existsSync(envFilePath)) {
-        throw new Error(`environment: ${envFilePath} does not exist`);
+        // Fallback to unsanitized filename for old envs
```

Comment:
could this be made to self repair the filename on first save?

Reply:
This looks great just had these two small notes. This is legitimately just a question, maybe self-repair here is going overboard since you are repairing the environment file contents.

Reply:
While it's certainly possible to fix/rename the filename upon saving. I think this would cause an unexpected side effect for the user.

Also, we would need a lot more logic to handle the rename. Because, renaming here would mean that the currently open file is “deleted” and a new file is “created”, causing the Tab to close.

Reply:
Yeah, makes sense. Probably not worth it, especially not in this already sizable change.